### ISSUE TITLE
youtube.com: breakage

### ISSUE THREAD

**Issue by ghost** [uBlockOrigin/uAssets#13465]
### Prerequisites

- [X] I read and understood the [policy as to what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets)
- [X] I verified that the issue has not already been reported (use [this button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out)
- [X] I forced an update of my filter lists (**how to do this:** [click _"Purge all caches"_](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches), then [click _"Update now"_](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now))
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/585534/91296476-ab9d8c00-e76a-11ea-9fb9-87d6861bdf73.png), or I verified that the issue is not caused by removing any of the default lists
- [X] I did not enable external filter lists, or I verified that the issue still occurs without enabling external filter lists
- [X] I do not have custom filters/rules, or I verified that the issue still occurs without custom filters/rules
- [X] I am not using [another content blocker along uBO](https://twitter.com/gorhill/status/1033706103782170625) (also mind browser built-in blockers)
- [ ] I did not answer truthfully to **all** the above checkpoints

### URL address of the web page

`https://www.youtube.com/watch?v=UwWFd_9fUYM&t=340s`

### Category

breakage

### Describe the issue

End screen cards not working when clicked

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![Screenshot 2022-05-30 182900](https://user-images.githubusercontent.com/106527668/170974603-3cd5bc67-c33d-4489-9d8d-b18dea9a6509.png)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.42.4
Chromium: 102
filterset (summary): 
  network: 157101
  cosmetic: 135525
  scriptlet: 36291
  html: 0
listset (total-discarded, last updated): 
  added: 
    https://winhelp2002.mvps.org/hosts.txt: 8730-0, 1d.3h.57m
    adguard-annoyance: 53588-1715, 1d.3h.56m
    adguard-generic: 64262-1071, 20h
    adguard-mobile: 7773-119, 19h.59m
    adguard-social: 17445-1945, 20h
    adguard-spyware: 26334-1041, 19h.59m
    adguard-spyware-url: 607-13, 19h.59m
    block-lan: 43-0, 7d.1h.8m
    curben-phishing: 13233-0, 19h.23m
    curben-pup: 250-0, 19h.22m
    dpollock-0: 10368-1440, 10d.40m
    fanboy-cookiemonster: 32254-16004, 5d.14h.34m
    fanboy-social: 23411-14175, 19h.59m
    fanboy-thirdparty_social: 68-21, 5d.14h.33m
    ublock-annoyances: 4431-214, 19h.59m
  default: 
    user-filters: 41-0, never
    easylist: 67598-25220, 19h.59m
    easyprivacy: 27110-10643, 19h.59m
    plowe-0: 3671-1481, 9d.1h.17m
    ublock-abuse: 75-0, 9d.12h.38m
    ublock-badware: 4139-308, 19h.59m
    ublock-filters: 31510-3895, 19h.59m
    ublock-privacy: 213-47, 9d.12h.37m
    ublock-quick-fixes: 189-14, 23h.54m
    ublock-unbreak: 1822-127, 19h.59m
    urlhaus-1: 9446-1, 18h.20m
filterset (user): [array of 41 redacted]
switchRuleset: 
  added: [array of 2 redacted]
modifiedUserSettings: 
  contextMenuEnabled: false
  ignoreGenericCosmeticFilters: true
  showIconBadge: false
  suspendUntilListsAreLoaded: true
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 2489 ms (selfie)
  maxAssetCacheWait: 2029 ms
popupPanel: 
  blocked: 17
  blockedDetails: 
    youtube.com: 9
    doubleclick.net: 2
    google.com: 2
    google.com.ph: 1
    googlevideo.com: 2
    ytimg.com: 1
```
</details>

**Comment by Branzeturi** [comment]
@mapx- But they still work on my side with AG Annoyances enabled

**Comment by Branzeturi** [comment]
Could you disable `"My Filters"` from `"Filter Lists`" (in uBlock Origin's dashboard?). Does that fix it?

Did anything hide any login prompt and/or any privacy-related annoyance?

**Comment by mapx-** [comment]
@Branzeturi did you test using chrome ?

**Comment by Branzeturi** [comment]
> @Branzeturi did you test using chrome ?

Yes

**Comment by Procyon-b** [comment]
@mapx-  I can replicate this problem with the default lists.
When I click the end cards, the player doesn't display the new video. Looking in the dev tools and opening yt with and without ubo doesn't show a noticeable difference in the "click" events.

**Comment by mapx-** [comment]
Weird. I can reproduce only in chrome + adguard annoyances.

**Comment by Procyon-b** [comment]
I've tried (now in FF and chrome) and clicking the end cards doesn't open the video.
```
uBlock Origin: 1.42.4
Firefox: 100
filterset (summary): 
  network: 83562
  cosmetic: 43618
  scriptlet: 17074
  html: 659
listset (total-discarded, last updated): 
  removed: 
    user-filters: null
  added: 
    block-lan: 43-0, 2m
  default: 
    ublock-filters: 32138-41, 3m
    ublock-badware: 4148-3, 3m
    ublock-privacy: 228-0, 3m
    ublock-abuse: 75-0, 2m
    ublock-unbreak: 1833-0, 2m
    easylist: 67664-598, 2m
    easyprivacy: 27114-579, 2m
    urlhaus-1: 9195-0, 2m
    plowe-0: 3669-3, 2m
    ublock-quick-fixes: 190-0, 2m
filterset (user): [array of 7 redacted]
modifiedUserSettings: 
  advancedUserEnabled: true
  showIconBadge: false
  webrtcIPAddressHidden: true
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 4975 ms
  maxAssetCacheWait: 2527 ms
popupPanel: 
  blocked: 12
  blockedDetails: 
    youtube.com: 6
    doubleclick.net: 2
    google.com: 1
    googlevideo.com: 2
    ytimg.com: 1
```

**Comment by gwarser** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/v020u4/issue_occurs_with_abp_and_ubo/

**Comment by Procyon-b** [comment]
yt/google could be experimenting with alternative designs.
I notice this when I have to work on my userscript that hides/dismiss login/cookie and other dialogs (disabled for the test here). Their javascript code/initialization is not always the same between page reload

**Comment by Procyon-b** [comment]
@gwarser your solution works for me in FF and chrome.
Branzeturi  should add that filter. `@@||www.youtube.com/generate_204$xhr,1p`

**Comment by mapx-** [comment]
so, in the end it's easyprivacy ?

**Comment by mapx-** [comment]
@ryanbr

**Comment by Branzeturi** [comment]
@Procyon-b I didn't encounter any issues with the use of `uBlock Origin` on `YouTube` today.

Maybe it's a regional thing

**Comment by Procyon-b** [comment]
at least for me it is.

**Comment by Procyon-b** [comment]
@Branzeturi possible
Sometimes I've difficulties to fix an issue for a user (of a userscript) that sees one type of cookie dialog, and I only see another one.

**Comment by Procyon-b** [comment]
@mapx- @gwarser this filter works too:
```
youtube.com##+js(no-fetch-if, /generate_204)
```

**Comment by mapx-** [comment]
Good to know, however maybe it should be fixed in EP, I added in quick just because of the high traffic

**Comment by Procyon-b** [comment]
I just had a rush of adrenaline because the browser where I last tested the filter wasn't experiencing the issue anymore (with no filter active). I had to go to the other computer to test it again.
I can confirm that `no-xhr-if` doesn't block the xhr request, but `no-fetch-if` does.

**Comment by Yuki2718** [comment]
I guess you mean add the rule to uBlock filters- Privacy. But TBH generate_204 is not much of a tracker as [commented](https://github.com/easylist/easylist/issues/7878#issuecomment-841886069) by @llacb47 .

**Comment by gwarser** [comment]
> @mapx- @gwarser this filter works too:
> 
> ```
> youtube.com##+js(no-fetch-if, /generate_204)
> ```

Are you 100% sure? I tried it before (redirections too) and it worked only for active tab. Did not work in new tab.

More likely, some `aeld` will work because it looks like a normal `a` link with `href`, but some event listener is breaking it.

**Comment by Procyon-b** [comment]
As I wrote, I had a doubt because commenting the rule made no difference (no problem anymore), but the browser was served a "normal" version of the page without any attempt to xhr that url. I had to start my second computer to check on that one.
I can try to run some more tests with background tabs if you want.

**Comment by gwarser** [comment]
I tried right now, and it does not work.

**Comment by Procyon-b** [comment]
I've re-tested on the second computer in FF and chrome. Opening the link in an active tab, in a background tab (right-click, ctrl + click "Open in new tab"/"Go to"), and pasting in the address bar of a new tab.
I have verified that without the `no-fetch-if` filter, clicking the card doesn't load the video. (the logger shows the blocked xhr request)
I then tried again with the filter active, and clicking worked. (no xhr event for "`https://www.youtube.com/generate_204`" logged)
